Source/Core/Core/HW/EXI/EXI.cpp

```diff
@@ -98,12 +136,13 @@ void Init()
     }
   }
 
-  for (int i = 0; i < MAX_MEMORYCARD_SLOTS; i++)
-    AddMemoryCards(i);
+  for (Slot slot : MEMCARD_SLOTS)
+    AddMemoryCards(slot);
```

Comment:
This is a preexisting issue but would it make sense to move the loop inside AddMemoryCards? The function is only called from here, and as it stands it adds a single device despite the name implying otherwise.

Reply:
I think that somewhat makes sense, but I'd rather rip out the entirety of `AddMemoryCards` and replace it with the new config system in a separate PR.  I can rename the function to `AddMemoryCard` pretty easily though.